### DISCUSSION TITLE
Proposal: Add Help Button to toolbar by default

### DISCUSSION THREAD

**Discussion by NicolaiRidani** [django-cms/django-cms#7094] [General]
Hi everybody, 

I have created a quick feature proposal. I would appreciate your feedback. 

**Situation**

The django CMS toolbar does not provide any sort of help button. As a user, I have to switch to my browser and search in Google for support or go directly to www.django-cms.org. 

**Solution**

Add a default help button to the django CMS toolbar that provides information about the various django CMS support resources (link to documentation, details about forum and Slack etc.) 

Mock up 1

BTW: The links in the drop down are not set in stone obviously...something we can work out together

![mockup1](https://user-images.githubusercontent.com/69975577/125792288-78e117e1-add7-479c-9697-0a969af88cc6.png)

Mock up 2

![mockup2](https://user-images.githubusercontent.com/69975577/125792297-f6474a9a-665c-4a20-a4a3-fea91be73d25.png)

What do you think? Which mock up do you prefer? Do you have other ideas?

**Comment by mahmudft** [comment]
Mock Up 1 looks great

**Comment by marksweb** [comment]
I quite like this idea with mockup 1 being the way I'd go about it 👍

**Comment by marksweb** [comment]
I found myself with some time tonight waiting for the kids to sleep & have had a go at how it could work;

https://github.com/marksweb/django-cms/commit/2182857bdb5ee9a8aa23966cfddf4d44022bf6e8

Only issues I had was adding `target="_blank"` to the links in the menu didn't open new tabs/windows so links would open in the same window currently. This is likely an issue with the way the toolbar is rendered in the page.

**Comment by NicolaiRidani** [comment]
Adding Nicolas' feedback from Slack to this discussion:

> Just my 0.02$! I'm not a 100% sure if I want that Help section there for my end users (the content editors) because most of the questions they've got are very content and project related. So the majority of users that use the CMS (not the admins) should not be redirected to open community support but to the project support. So I would appreciate if that help section can be toggled on or off for certain user groups.

I think making it easy to switch off makes absolutely sense. @evildmp also suggested in today's weekly Tech Committee meeting that the help section could also be amended by adding help information / project support details of the project owner.

**Reply by marksweb** [reply]
Funny those things should come up @NicolaiRidani

In that initial approach I added, the toolbar can be turned off be a setting and you can change what appears in it.

If we wanted to ensure some cms links were there & there was an option for people to add others then that'd also be fairly simple to do 👍

To provide some detail here as to what my branch contains, I'd added the settings;

```
    'ENABLE_HELP': True,  # Adds help menu toolbar
    'HELP_MENU_ITEMS': (
        (_('Community forum'), 'https://discourse.django-cms.org/'),
        (_('Documentation'), 'https://docs.django-cms.org/en/latest/'),
        (_('Getting started'), 'https://www.django-cms.org/en/get-started-django-cms/'),
        (_('Talk to us'), 'https://www.django-cms.org/en/support/'),
    )
}
```

Then the menu is added if it's enabled and populated with the contents of that `HELP_MENU_ITEMS` setting;

```
    def add_help_menu(self):
        """ Adds the help menu if it's enabled in settings """
        if get_cms_setting('ENABLE_HELP'):
            self._help_menu = self.toolbar.get_or_create_menu(HELP_MENU_IDENTIFIER, _('Help'), position=-1)
            menu_items = get_cms_setting('HELP_MENU_ITEMS')
            for label, url in menu_items:
                self._help_menu.add_link_item(label, url=url)
```

This was a simple initial implementation to demo what it might do, but we could make the menu content a bit more advanced - having links we enforce and optional user provided links etc.

**Comment by goutnet** [comment]
Should we create an issue for this? (and add it to a milestone?)

**Reply by marksweb** [reply]
Yeah that could be a good move. I could create a draft PR against it from what I've done so far, then hopefully get some feedback on things.

**Comment by NicolaiRidani** [comment]
Links to be included by default:

- Getting started developer guide (link: https://docs.django-cms.org/en/latest/introduction/index.html)
- Documentation (link: https://docs.django-cms.org/en/latest/index.html)
- Support Forum (link: https://discourse.django-cms.org/)
- Support Slack (link: https://www.django-cms.org/slack)
- What's new (link: https://www.django-cms.org/en/blog/) 


I would also like to add a "Getting Started User Guide" in the future once the tutorial is completed. 

What do you think of the link (+labels)?

**Reply by marksweb** [reply]
Yeah that works 👍

Will put in place when I get chance
